google/resource_kms_crypto_key.go

```diff
@@ -42,6 +43,16 @@ func resourceKmsCryptoKey() *schema.Resource {
 	}
 }
 
+func kmsCryptoKeyRingsEquivalent(k, old, new string, d *schema.ResourceData) bool {
+	keyRingIdWithSpecifiersRegex := regexp.MustCompile("^projects/([a-z0-9-]+)/locations/([a-z0-9-])+/keyRings/([a-zA-Z0-9_-]{1,63})$")
```

Comment:
Rely on `parseKmsKeyRingId` method instead. Parse the new with that method and compare the value with the old  value which should be canonical.

Reply:
Can't - that requires a `Config` object not available at `DiffSuppressFunc` time.